src/proxy/memory.js

```diff
@@ -26,6 +28,17 @@ module.exports = function createMemoryProxy(config) {
 		return -1;
 	}
 
+	function castId(type, id) {
+		switch(type) {
+			case "string":
+				break;
+			case "number":
+				id = parseFloat(id);
```

Comment:
It should be int or string. Should check if the type is number or not. Also, you should check the type on the string side and the conversion as well.